src/main/java/edu/greenblitz/tobyDetermined/subsystems/LED.java

```diff
@@ -31,45 +30,51 @@ public static void init(){
         }
     }
 
-    public void setColor(Color color) {
-        for (int i = 0; i < this.ledBuffer.getLength(); i++) {
-            this.ledBuffer.setLED(i, color);
-            SmartDashboard.putNumber("led num", i);
-        }
-        this.addressableLED.setData(ledBuffer);
-    }
     public void setColor(int i, Color color) {
         this.ledBuffer.setLED(i, color);
-        this.addressableLED.setData(ledBuffer);
     }
 
+    public void setColor (int startIndex,int endIndex, Color color){
+        for (int i = startIndex; i < endIndex; i++) {
+            setColor(i,color);
+        }
+    }
+
+    public void setColor (RobotMap.LED.Sections section, Color color){
+        setColor(section.start,section.end,color);
+    }
 
     public void turnOff (){
-        setColor(new Color(0,0,0));
+        setColor(RobotMap.LED.Sections.ALL,new Color(0,0,0));
     }
 
     public void turnOff (int index){
         setColor(index,new Color(0,0,0));
     }
-    public void turnoff (int startIndex,int endIndex){
+    public void turnOff (int startIndex,int endIndex){
         for (int i = startIndex; i < endIndex; i++) {
-            setColor(i,new Color(0,0,0));
+            turnOff(i);
         }
     }
+    public void turnOff (RobotMap.LED.Sections section){
+        turnOff(section.start,section.end);
+    }
 
 
 
     public void setHSV(int h, int s, int v) {
         for (int i = 0; i < this.ledBuffer.getLength(); i++) {
             this.ledBuffer.setHSV(i, h, s, v);
         }
-        this.addressableLED.setData(ledBuffer);
     }
 
 
     public void setHSV(int index, int h, int s, int v) {
         this.ledBuffer.setHSV(index, h, s, v);
-        this.addressableLED.setData(ledBuffer);
     }
 
+    @Override
+    public void periodic() {
```

Comment:
I'd like to hear why did you choose to update every cycle instead of when the values are updated.